pkg/action/install.go

```diff
@@ -584,14 +592,107 @@ func (c *Configuration) renderResources(ch *chart.Chart, values chartutil.Values
+	if pr == nil {
+		return files, nil
+	}
+
+	annotationName := "helmPostProcessFileName"
+
+	b := bytes.NewBuffer(nil)
+
+	// Split the rendered files into documents and add the temp annotation.
+	reYamlDocumentSeparator := regexp.MustCompile(`\n---\s*\n`)
+	for fileName, fileContent := range files {
+		for _, document := range reYamlDocumentSeparator.Split(fileContent, -1) {
+			if strings.TrimSpace(document) == "" {
+				continue
+			}
+			b.WriteString("---\n")
+			documentYaml, _, err := yamlSetTemporaryAnnotation(document, annotationName, fileName)
+			if err != nil {
+				return nil, fmt.Errorf("cannot prepare a file for post-process: %w", err)
+			}
+			b.Write(documentYaml)
+		}
+	}
+
+	// Run post-processing.
+	b, err := pr.Run(b)
+	if err != nil {
+		return nil, errors.Wrap(err, "error while running post render on files")
+	}
+
+	postProcessedFiles := make(map[string]string)
+	// Split the post-processed stream into files.
+	for _, document := range strings.Split(b.String(), "---\n") {
+		if strings.TrimSpace(document) == "" {
+			continue
+		}
+		document, fileName, err := yamlSetTemporaryAnnotation(document, annotationName, "")
 		if err != nil {
-			return hs, b, notes, errors.Wrap(err, "error while running post render on files")
+			return nil, fmt.Errorf("cannot reconstruct a file after post-process: %w", err)
+		}
+		existingDocument, ok := postProcessedFiles[fileName]
+		if ok {
+			postProcessedFiles[fileName] = existingDocument + "\n---\n" + string(document)
+		} else {
+			postProcessedFiles[fileName] = string(document)
 		}
 	}
 
-	return hs, b, notes, nil
+	return postProcessedFiles, nil
+}
+
+// yamlSetTemporaryAnnotation adds or cuts out an annotation text in yaml kubernetess object
+func yamlSetTemporaryAnnotation(document, annotationName, annotationValue string) ([]byte, string, error) {
+	// Unmarshal the document.
+	var documentMap map[string]interface{}
+	err := yaml.Unmarshal([]byte(document), &documentMap)
+	if err != nil {
+		return nil, "", err
+	}
+
+	// Take the annotations map.
+	metadata, ok := documentMap["metadata"].(map[string]interface{})
+	if !ok {
+		return nil, "", fmt.Errorf("document has no metadata: %s", document)
+	}
+
+	annotations, ok := metadata["annotations"].(map[string]interface{})
+
+	// Add or remove the annotation.
+	if annotationValue == "" && ok { // Remove
+		annotationValue, ok = annotations[annotationName].(string)
+		if !ok {
+			return nil, "", fmt.Errorf("annotation %s value is not a string: %v", annotationName, annotations[annotationName])
+		}
+		delete(annotations, annotationName)
+	} else if annotationValue != "" { // Add
+		if !ok {
+			annotations = make(map[string]interface{})
+		}
+		annotations[annotationName] = annotationValue
+	} else { // Nothing to remove.
+		return nil, "", fmt.Errorf("annotation %s is not set and therefore cannot be deleted", annotationName)
```

Comment:
When splitting this out to its own function for removing, you probably don't need to return an error here. I don't think it indicates any real problem if there is no annotation to remove

Reply:
I already shoot myself in the foot because of that error, when Kustomize post-renderer added a new networkPolicy object  :)